tests/rootfs.sh

```diff
@@ -0,0 +1,149 @@
+#!/bin/sh
+
+set -eu
+
+ERROR=1
+OK=0
+MY_NAME=$(basename -- "$0")
+ROOTFS_DIRECTORY="./rootfs"
+TEST_IMAGE_FILE_PATH="${ROOTFS_DIRECTORY}/tmp/test.img"
+TEST_EMPTY_MTAB_FILE="${ROOTFS_DIRECTORY}/etc/mtab"
+
+test_setup()
+{
+    if [ ! -f "${ROOTFS_DIRECTORY}/usr/bin/qemu-arm-static" ]
+    then
+        cp "$(command -v qemu-arm-static)" "${ROOTFS_DIRECTORY}/usr/bin/"
+    fi
+
+    dd if=/dev/zero of="${TEST_IMAGE_FILE_PATH}" bs=2M count=64 >/dev/null 2>&1 || return ${ERROR}
+    touch "${TEST_EMPTY_MTAB_FILE}" >/dev/null 2>&1 || return ${ERROR}
+    mkfs.ext4 "${TEST_IMAGE_FILE_PATH}" >/dev/null 2>&1 || return ${ERROR}
+    mount -o bind /dev/ "${ROOTFS_DIRECTORY}/dev/" >/dev/null 2>&1 || return ${ERROR}
+}
+
+test_destroy()
+{
+    umount -l "${ROOTFS_DIRECTORY}/dev"
+
+    if [ -f "${TEST_IMAGE_FILE_PATH}" ]
+    then
+        rm -f ${TEST_IMAGE_FILE_PATH}
+    fi
+
+    if [ -f "${TEST_EMPTY_MTAB_FILE}" ]
+    then
+        rm -f "${TEST_EMPTY_MTAB_FILE}"
+    fi
+    if [ -f "${ROOTFS_DIRECTORY}/usr/bin/qemu-arm-static" ]
+    then
+        rm -f "${ROOTFS_DIRECTORY}/usr/bin/qemu-arm-static"
+    fi
+}
+
+run_test()
+{
+    if ! test_setup
+    then
+        printf "Cannot run tests, unable to complete test setup\\n"
+        test_destroy
+        exit ${ERROR}
```

Comment:
and here we are in a jiffie sadly functions do not return anything, they set exit codes. so how do we get the exit code in the variable without using a global variable. (if we are using a global var, why return anything anyway).

so RESULT=test_setup doesn't exist. what happens if we capture the exit code, e.g. RESULT=$(test_setup) sub-shell execution, i guess it won't try to run a local function .. hmm, while the global var works, it's also ambiguous.
the only thing come up with is
```sh
test_setup
RESULT="${?}"
if [ "${RESULT}" -ne 0 ]; then
    exit "${RESULT}"
fi
```
which is a bit equally ugly, but the lesser of the two evils. imrpovements welcome

Reply:
The return statement sets the exit code of the function call, this can be interpreted by the if statement without using the '[' program and therefore works. e.g.: 

> if function; then ....